invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalUsageInArrear.java

```diff
@@ -557,4 +587,160 @@ public LocalDate getNextNotificationDate() {
         }
     }
+
+    public BigDecimal toBeBilledForUnit(List<UsageInArrearDetail> toBeBilledDetails){
+        BigDecimal result = BigDecimal.ZERO;
+        for (UsageInArrearDetail toBeBilled: toBeBilledDetails){
+            result = result.add(toBeBilled.getAmount());
+        }
+        return result;
+    }
+
+    private List<UsageInArrearDetail> reconcileExistedBilledWithToBeBilled(Iterable<InvoiceItem> billedItems, List<UsageInArrearDetail> toBeBilledUsageInArrearDetails) {
+        for (final InvoiceItem bi : billedItems) {
+            List<UsageInArrearDetail> billedUsageItemDetails = fromJson(bi.getItemDetails());
+
+            if (billedUsageItemDetails != null && billedUsageItemDetails.size() > 0) {
+
+                for (final UsageInArrearDetail toBeBilledConsumable : toBeBilledUsageInArrearDetails) {
+                    billedUsageItemDetails = toBeBilledConsumable.reconcile(billedUsageItemDetails);
+                }
+
+                if (billedUsageItemDetails != null && billedUsageItemDetails.size() > 0) {
+                    for (final UsageInArrearDetail billedUsage : billedUsageItemDetails) {
+                        toBeBilledUsageInArrearDetails.add(new UsageInArrearDetail(billedUsage.getTier(), billedUsage.getTierUnit(), billedUsage.getTierPrice(),
+                                                                                   billedUsage.getQuantity(), billedUsage.getAmount().negate(), null, bi.getId().toString()));
+                    }
+                }
+            } else {
+                toBeBilledUsageInArrearDetails.add(new UsageInArrearDetail(bi.getRate(), bi.getQuantity(), bi.getAmount().negate(), bi.getId().toString()));
+            }
+        }
+
+        return toBeBilledUsageInArrearDetails;
+    }
+
+    private static final String toJson(List<UsageInArrearDetail> toBeBilledUsageInArrearDetails) {
+        String result = null;
+        if (toBeBilledUsageInArrearDetails != null && toBeBilledUsageInArrearDetails.size() > 0){
+            try {
+                result = objectMapper.writeValueAsString(toBeBilledUsageInArrearDetails);
+            } catch (JsonProcessingException e) {
+                Preconditions.checkState(false, e.getMessage());
+            }
+        }
+        return result;
+    }
+
+    private static final List<UsageInArrearDetail> fromJson(String itemDetails){
+        List<UsageInArrearDetail> toBeBilledUsageInArrearDetails = null;
+        if (itemDetails != null){
+            try {
+                toBeBilledUsageInArrearDetails = objectMapper.readValue(itemDetails, new TypeReference<List<UsageInArrearDetail>>() {});
+            } catch (IOException e) {
+                Preconditions.checkState(false, e.getMessage());
+            }
+        }
+
+        return toBeBilledUsageInArrearDetails;
+    }
+
+    // divide the price between all details to prevent a detail to be removed if the price is zero when the usage mode is detail
```

Comment:
We cannot do that: even in `detail` mode, a single entry would be returned for CAPACITY. See also my comment at https://github.com/killbill/killbill-integration-tests/pull/13#discussion_r166228035.